cpu/esp32/esp-ieee802154/esp_ieee802154_hal.c

```diff
@@ -0,0 +1,466 @@
+        return esp_ieee802154_clear_pending_addr(value, true) ? -ENOENT : 0;
+    case IEEE802154_SRC_MATCH_EXT_ADD:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_EXT_ADD\n", __func__);
+        return esp_ieee802154_add_pending_addr(value, false) ? -ENOMEM : 0;
+    case IEEE802154_SRC_MATCH_EXT_CLEAR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_EXT_CLEAR\n", __func__);
+        return esp_ieee802154_clear_pending_addr(value, false) ? -ENOENT : 0;
+    }
+
+    return -EINVAL;
+}
+
+void esp_ieee802154_cca_done(bool channel_busy)
+{
+    DEBUG("[esp_ieee802154] %s %d\n", __func__, channel_busy);
+    _channel_clear = !channel_busy;
+
+    if (IS_ACTIVE(_USE_CCA_DONE)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_CCA);
+    }
+}
+
+void esp_ieee802154_receive_sfd_done(void)
+{
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+    if (IS_ACTIVE(_USE_RX_START)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_RX_START);
+    }
+}
+
+void esp_ieee802154_receive_done(uint8_t *frame, esp_ieee802154_frame_info_t *frame_info)
+{
+    assert(frame);
+    assert(frame_info);
+
+    DEBUG("[esp_ieee802154] %s: frame=%p frame_info=%p\n",
+          __func__, frame, frame_info);
+
+    _rx_frame = frame;
+    _rx_frame_info = frame_info;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_RX_DONE);
+}
+
+void esp_ieee802154_transmit_sfd_done(uint8_t *frame)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p\n", __func__, frame);
+    if (IS_ACTIVE(_USE_TX_START)) {
+        esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_INDICATION_TX_START);
+    }
+}
+
+void esp_ieee802154_transmit_done(const uint8_t *frame, const uint8_t *ack,
+                                  esp_ieee802154_frame_info_t *ack_frame_info)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p ack_frame=%p ack_frame_info=%p\n",
+          __func__, frame, ack, ack_frame_info);
+
+    _ack_frame = ack;
+    _ack_frame_info = ack_frame_info;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+}
+
+void esp_ieee802154_transmit_failed(const uint8_t *frame, esp_ieee802154_tx_error_t error)
+{
+    DEBUG("[esp_ieee802154] %s: frame=%p error=%u\n", __func__, frame, error);
+
+    _tx_error = error;
+
+    esp_ieee802154_dev->cb(esp_ieee802154_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+}
+
+void esp_ieee802154_energy_detect_done(int8_t power)
+{
+    DEBUG("[esp_ieee802154] %s: power %i\n", __func__, power);
+}
+
+void esp_ieee802154_setup(ieee802154_dev_t *dev)
+{
+    assert(dev);
+
+    DEBUG("[esp_ieee802154] %s: dev=%p\n", __func__, dev);
+
+    dev->driver = &esp_ieee802154_driver;
+    esp_ieee802154_dev = dev;
+}
+
+int esp_ieee802154_init(void)
+{
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+
+    esp_ieee802154_enable();
+    esp_ieee802154_reset_pending_table(true);
+    esp_ieee802154_reset_pending_table(false);
```

Comment:
Why is this called twice?

Reply:
The parameter tells what source matching table is cleared (true for short addressed and false for extended addresses):
```c
/**
 * @brief  Clear the source matching table to empty.
 *
 * @param[in]  is_short  Clear Short address table or Extended address table.
 *
 * @return
 *      - ESP_OK on success.
 *      - ESP_FAIL on failure.
 */
esp_err_t esp_ieee802154_reset_pending_table(bool is_short);
```